### ISSUE TITLE
Configuration cache support: Task.project in AssertGraphTask 

### ISSUE THREAD

**Issue by eugene-krivobokov** [jraska/modules-graph-assert#120]
Scenario:
`./gradlew assertModuleGraph --configuration-cache`

```
org.gradle.api.InvalidUserCodeException: Invocation of 'Task.project' by task ':assertModuleLayers' at execution time is unsupported.
...
  at org.gradle.api.DefaultTask.getProject(DefaultTask.java:59)
  at com.jraska.module.graph.assertion.tasks.AssertGraphTask.run(AssertGraphTask.kt:20)
```

https://docs.gradle.org/current/userguide/configuration_cache.html#config_cache:requirements:use_project_during_execution

Environment:
- Gradle 7.2
- Plugin version: 1.5.1

**Comment by jraska** [comment]
So I looked into it and the problem is this: https://docs.gradle.org/7.2/userguide/configuration_cache.html#config_cache:requirements:use_project_during_execution
Accessing `Project` object - the thing is this is the main feature of this plugin and we have to do this to get the module dependencies graph and to do assertions on it.

If we cannot do this we have to do it at configuration time, which will require also changing of all the implementations for the asserts - though this seems desirable also with relation to #38 

I see this for version 2.0 actually so it might take some time to support configuration cache and currently I'm not even sure if it is possible given literally all we do is accessing the modules - `Project`s.

**Comment by eugene-krivobokov** [comment]
> we have to do this to get the module dependencies graph and to do assertions on it

I thought there are might be alternative ways to avoid this manual resolving:

- Put the logic inside dependency resolution strategy:

```kotlin
configuration.resolutionStrategy.eachDependency {
  if (isDependencyForbidden(project, requestedDependency)) throwError()
}
```

- Add "fake" configuration with version constraint `rejectAll`: 
https://gradle-community.slack.com/archives/CAH4ZP3GX/p1629204946070200

These are only ideas to check.

**Comment by jraska** [comment]
Resolved by [Release 2.0.0](https://github.com/jraska/modules-graph-assert/releases/tag/2.0.0)

Please give it a try and reopen the issue in case something doesn't work.
